### PR TITLE
chore: change streaming upload timeout to 20s only for backfill test

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
As title, wait for https://github.com/risingwavelabs/risingwave/pull/18693

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/connector/src/parser/protobuf/recursive.rs
Comment (MrCroxx) [reply src/connector/src/parser/protobuf/recursive.rs]: Wired... Why the phantom is keep popping up. = =

File: Cargo.toml
Comment (hzxa21) [inline Cargo.toml:81]: Revert this change after #18693?

Comment (wcy-fdu) [reply Cargo.toml:81]: Yes, will merge main after #18693 merged.

File: src/config/ci.toml
Comment (hzxa21) [inline src/config/ci.toml:20]: This change will affect other tests, not just backfill test.

Comment (wcy-fdu) [reply src/config/ci.toml:20]: Yes, but if this change is introduced to other tests, what will be the impact? I think they won’t break even if timeout is increased, even for aws sdk. Main reason I suggest this is to reduce maintenance workload

Comment (hzxa21) [reply src/config/ci.toml:20]: The impact will be if there is any performance regression in the sdk, the chance we can catch it in CI will be lowered. For example, if for some reason the sdk is 10s slower in a new version in some corner cases, the CI test will still pass, while before this change, the CI test will fail.

There are four profiles used in `run-backfill-tests.sh`:
1. ci-3cn-1fe
2. ci-3cn-1fe-with-minio-rate-limit
3. ci-3cn-1fe-with-monitoring
4. ci-3cn-1fe-with-monitoring-and-minio-rate-limit

1 is used by many tests and 2-4 are used by backfill only. I think we can:
- Add one more toml files copied from `ci.toml` and change the timeout only.
- Change the 2-4 prefix from `ci` to `ci-backfill` and add one more risedev profile `ci-backfill-3cn-1fe`.
- Let the backfill test use the new profiles with the new toml config.

**Review by hzxa21 — APPROVED** [review]
Rest LGTM

**Pull request merged**
